File: PYTHON/parking_lot.py
class ParkingLot:
    def __init__(self, kapasitas=""):
        self._kapasitas = kapasitas
        self._jumlah_kendaraan_saat_ini = 0

    @property
    def kapasitas(self):
        return self._kapasitas

    @property
    def jumlah_kendaraan_saat_ini(self):
        return self._jumlah_kendaraan_saat_ini

    def tambah_kendaraan(self):
        if self._jumlah_kendaraan_saat_ini < self.kapasitas:
            self._jumlah_kendaraan_saat_ini += 1
        else:
            print("Tempat parkir penuh.")

File: CPP/parking_lot.cpp
#pragma once

#include <string>
#include <vector>
#include <iostream>
#include "vehicle.cpp"

class ParkingLot {
private:
    int _kapasitas;
    int _jumlah_kendaraan_saat_ini;

public:
    ParkingLot(int kapasitas = 0);
    
    int getKapasitas();
    int getJumlahKendaraanSaatIni();
    
    void tambahKendaraan();
};

ParkingLot::ParkingLot(int kapasitas)
    : _kapasitas(kapasitas), _jumlah_kendaraan_saat_ini(0) {}

int ParkingLot::getKapasitas() {
    return _kapasitas;
}

int ParkingLot::getJumlahKendaraanSaatIni() {
    return _jumlah_kendaraan_saat_ini;
}

void ParkingLot::tambahKendaraan() {
    if (_jumlah_kendaraan_saat_ini < _kapasitas) {
        _jumlah_kendaraan_saat_ini++;
    } else {
        std::cout << "Tempat parkir penuh." << std::endl;
    }
}
